src/symbol/placement.js

```diff
@@ -399,7 +422,38 @@ export class Placement {
             bucket.updateCollisionDebugBuffers(this.transform.zoom, collisionBoxArray);
         }
 
+        const getSymbolFeature = (symbolInstance: SymbolInstance): VectorTileFeature => {
+            assert(clippingData);
+            const featureIndex = clippingData.featureIndex;
+            const retainedQueryData = this.retainedQueryData[bucket.bucketInstanceId];
+            return featureIndex.loadFeature({
```

Comment:
I haven't profiled, but I suspect this line to be an expensive part of the filtering, did you find this to be acceptable? Otherwise, is there any way to eventually cache/hold the feature data in the clippingData?

I'm mainly concerned about these statements to be called on placement frames:
```
this.loadVTLayers();
const sourceLayerName = this.sourceLayerCoder.decode(sourceLayerIndex);
const sourceLayer = this.vtLayers[sourceLayerName];
const feature = sourceLayer.feature(featureIndex);
```

Reply:
good call, I added some caching to cache the result of deserialization of `VectorTileFeature`